### PR TITLE
Integer overflow 4516 v4

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/4516

Describe changes:
- Avoids some warnings about integers

Obviously not complete yet... But still mergeable ?

Replaces #6580 with
- right negation for improved bool definition

libhtp-pr: 338

### CODE REVIEW

File: .github/workflows/builds.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ env:
   DEFAULT_SV_BRANCH: master
   DEFAULT_SV_PR:
 
-  DEFAULT_CFLAGS: "-Wall -Wextra -Werror -Wno-unused-parameter -Wno-unused-function"
+  DEFAULT_CFLAGS: "-Wall -Wextra -Werror -Wno-unused-parameter -Wno-unused-function -Wimplicit-int-float-conversion"
```
Comment (catenacyber) [inline .github/workflows/builds.yml:20]: @jasonish where should I put this ?
This flag is not supported on the CentOS 8 build

Comment (catenacyber) [reply .github/workflows/builds.yml:20]: cc @ct0br0

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/6628

**Comment by jasonish** [comment]
> @jasonish where should I put this ? This flag is not supported on the CentOS 8 build

It will have to be added to the builds that support it, and not the default cflags.

**Pull request closed without merging**